crates/holochain/src/core/workflow/publish_dht_ops_workflow.rs

```diff
@@ -63,26 +67,28 @@ pub async fn publish_dht_ops_workflow(
             Err(e) => {
                 // If we get a routing error it means the space hasn't started yet and we should try publishing again.
                 if let holochain_p2p::HolochainP2pError::RoutingDnaError(_) = e {
+                    // TODO if this doesn't change what is the loop terminate condition?
```

Comment:
As a general comment, there is no circuit breaker to prevent this from happening repeatedly. In general using `WorkflowComplete::Incomplete` is quite dangerous for errors that are expected to be temporary. I'd actually vote for removing the concept entirely. The workflow should do what it needs to do then wait for a timer or another event. If you need to be called again immediately then you should be able to do the same work right away.

Reply:
Yeah, we could remove that whole concept. I'm not sure about waiting for a timer though. If you just mean adding a small delay in between iterations when one pass through is not enough, then I agree. But less in agreement if you mean repeatedly calling the workflow on a timer. If everything is hooked up properly, then triggers should be sufficient, and a loop would just hide the fact that things aren't hooked up properly.

Reply:
That feels good to me. The app validation workflow functions similarly. When it fails it should be retriggered. But that's not efficient or even very logical, to hope that "things will be better next time around". What could make sense is to retrigger another workflow before the current one, but even that is based on hope.

Reply:
Yes I'm not talking about increasing timer frequency or new timers. I'm saying we should rely on triggers when something changes in the system or an existing timer triggers the workflow and attempt to refactor out the "retrigger self" conditions. So in this example, if the cell hasn't finished joining the network when this workflow runs then the trigger is probably wrong and we should fix that rather than scheduling ourselves to run again and risking a loop if the network join isn't proceeding.

A long way of saying I agree, we should iron out issues with triggers and not start randomly adding loops until things work :)